### PR TITLE
Feature/INA-137

### PR DESCRIPTION
**Pull request by aehrenthal**
This pull request includes additional hub API improvements to make the user account creation process more seamless.

### CODE REVIEW

File: hub/hub-api.yaml [hunk 1]
```diff
@@ -825,6 +705,30 @@ paths:
                 $ref: "#/components/schemas/ResponseError"
 
   /users:
+    post:
+      tags:
+        - Users
+      summary: Create a new user
+      description: Use this endpoint to add a new user to Inperium. Provide basic information about the new user. They will receieve an invitation by email.
```
Comment (BullDog57Rus) [inline hub/hub-api.yaml:712]:
```suggestion
      description: Use this endpoint to add a new user to Inperium. Provide basic information about the new user.
```
it is not only invite

File: hub/hub-api.yaml [hunk 2]
```diff
@@ -825,6 +705,30 @@ paths:
...
+      operationId: createUser
+      requestBody:
+        content:
+          "*/*":
```
Comment (BullDog57Rus) [inline hub/hub-api.yaml:716]:
```suggestion
          application/json:
```

File: hub/hub-api.yaml [hunk 3]
```diff
@@ -1566,6 +1470,7 @@ components:
                 - TWO_FACTOR_AUTH_TOKEN_INVALID
                 - EMAIL_VERIFICATION_REQUIRED
                 - REFRESH_TOKEN_EXPIRED
+                - INSUFFICIENT_PERMISSIONS
```
Comment (BullDog57Rus) [inline hub/hub-api.yaml:1473]: where do u need it? I think 403 Forbidden in made for this :)

File: hub/hub-api.yaml [hunk 4]
```diff
@@ -2067,60 +2021,20 @@ components:
...
       description: Model for changing user password
       required:
-        - passwordResetToken
+        - email
+        - oneTimePassword
         - confirmPassword
         - newPassword
       properties:
-        passwordResetToken:
+        email:
```
Comment (BullDog57Rus) [inline hub/hub-api.yaml:2029]: why do we need both email and oneTimeCode? the email to find the user?

File: hub/hub-api.yaml (hunk 3 above)
Comment (BullDog57Rus) [reply hub/hub-api.yaml:1473]: I just think that currently, we add this if the FE needs to know anything about the exception

File: hub/hub-api.yaml [hunk 5]
```diff
@@ -1624,6 +1529,27 @@ components:
...
+      required:
+        - email
+      properties:
+        email:
+          type: string
+        firstName:
+          type: string
+        lastName:
+          type: string
+        isAdmin:
```
Comment (BullDog57Rus) [inline hub/hub-api.yaml:1544]:
```suggestion
        admin:
```
we have a boolean getter prefix "is", so not to have isIsAdmin we can do this :)

File: hub/hub-api.yaml (hunk 2 above)
Comment (aehrenthal) [reply hub/hub-api.yaml:716]: Done

File: hub/hub-api.yaml (hunk 3 above)
Comment (aehrenthal) [reply hub/hub-api.yaml:1473]: Done

File: hub/hub-api.yaml (hunk 4 above)
Comment (aehrenthal) [reply hub/hub-api.yaml:2029]: Just for security, that way you need to know the user email + their oneTimeCode and cannot just "guess" it.

File: hub/hub-api.yaml (hunk 5 above)
Comment (aehrenthal) [reply hub/hub-api.yaml:1544]: I wanted to change this one too, but sell and the starter library use this one too and I wanted to keep my changes in this PR as small as possible (given that its already quite big)

File: hub/hub-api.yaml [hunk 6]
```diff
@@ -1624,6 +1528,62 @@ components:
...
+            lastName:
+              type: string
+            isAdmin:
+              type: boolean
+            roles:
+              type: array
+              items:
+                type: string
+            tenantId:
+              $ref: "#/components/schemas/Id"
```
Comment (RockMother) [inline hub/hub-api.yaml:1570]: Not sure that we really need to use tenantId here. Probably it's a potential security issue.

**Review by RockMother — CHANGES_REQUESTED** [review]

File: hub/hub-api.yaml (hunk 6 above)
Comment (aehrenthal) [reply hub/hub-api.yaml]: We just removed it, thanks. It was checked in the code, but it's still safer not to expose it on the API.

**Review by BullDog57Rus — APPROVED** [review]

**Pull request merged**